src/EFCore/ChangeTracking/CollectionEntry.cs

```diff
@@ -82,6 +82,108 @@ public new virtual IEnumerable CurrentValue
             [param: CanBeNull] set => base.CurrentValue = value;
         }
 
+        /// <summary>
+        ///     Gets or sets a value indicating whether any of foreign key property values associated
+        ///     with this navigation property have been modified and should be updated in the database
+        ///     when <see cref="DbContext.SaveChanges()" /> is called.
+        /// </summary>
+        public override bool IsModified
+        {
+            get
+            {
+                var stateManager = InternalEntry.StateManager;
+
+                if (Metadata is ISkipNavigation skipNavigation)
+                {
+                    if (InternalEntry.EntityState != EntityState.Unchanged
+                        && InternalEntry.EntityState != EntityState.Detached)
+                    {
+                        return true;
+                    }
+
+                    var joinEntityType = skipNavigation.JoinEntityType;
+                    var foreignKey = skipNavigation.ForeignKey;
+                    foreach (var joinEntry in stateManager.Entries)
```

Comment:
```suggestion
                    foreach (var joinEntry in stateManager.GetEntriesForState(added: true, modified: true, deleted: true, unchanged: false))
```

And remove the redundant checks below

Reply:
All states are now needed.